analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ObjectsShouldNotBeDisposedMoreThanOnce.vb

```diff
@@ -0,0 +1,242 @@
+﻿Imports System
+Imports System.IO
+Imports System.Data.Common
+Imports System.Threading.Tasks
+
+Interface IWithDispose
+    Inherits IDisposable
+End Interface
+
+Public Class Disposable
+    Implements IDisposable
+
+    Public Sub Dispose() Implements IDisposable.Dispose
+    End Sub
+
+End Class
+
+Public Class DisposableAlias
+    Implements IDisposable
+
+    Public Sub CleanUp() Implements IDisposable.Dispose
+    End Sub
+
+End Class
+
+Public Class DoesNotImplementDisposable
+
+    Public Sub Dispose()
+    End Sub
+
+End Class
+
+Class Program
+
+    Public Sub NotReallyDisposable()
+        Dim d = New DoesNotImplementDisposable()
+        d.Dispose()
+        d.Dispose() ' Noncompliant^9#11 - IDisposal interface implementation is not checked
```

Comment:
I didn't find any case like structs (where we can have a Dispose() without implementing the IDisposable).

Not sure if I should check the interface implementation here.

Reply:
Ref structs do not exist in VB

Reply:
Yes - my question was if there is a case in VB similar to ref structs that I'm not aware of.

And independently do we want here to check if `Dispose` comes from the IDisposable interface? (IDisposableAsync is checked by the `KnownMethods.IsIAsyncDisposableDisposeAsync(this IMethodSymbol methodSymbol)` changed in previous PR).